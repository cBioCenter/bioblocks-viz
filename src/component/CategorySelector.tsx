import * as React from 'react';
import { Dropdown } from 'semantic-ui-react';

export interface ICategorySelectorProps {
  /** Categories to select from. */
  categories: string[];
  /** Callback for when a new category is selected. */
  onCategoryChange?: (event: React.SyntheticEvent<any>, data: object) => void;
}

/**
 * Class to represent a dropdown.
 *
 * @extends {React.Component<ICategorySelectorProps, any>}
 */
export default class CategorySelector extends React.Component<ICategorySelectorProps, any> {
  constructor(props: ICategorySelectorProps) {
    super(props);
  }

  public render() {
    return (
      this.props.categories && (
        <Dropdown
          fluid={true}
          onChange={this.props.onCategoryChange}
          options={[
            { key: 'all', text: 'all', value: undefined },
            ...this.props.categories.map(cat => {
              return { key: cat, text: cat, value: cat };
            }),
          ]}
          placeholder={'Select Category'}
          search={true}
        />
      )
    );
  }
}

export { CategorySelector };
