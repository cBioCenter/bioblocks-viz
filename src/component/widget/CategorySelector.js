"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var React = require("react");
var semantic_ui_react_1 = require("semantic-ui-react");
/**
 * Class to represent a dropdown.
 *
 * @extends {React.Component<ICategorySelectorProps, any>}
 */
var CategorySelector = /** @class */ (function (_super) {
    tslib_1.__extends(CategorySelector, _super);
    function CategorySelector(props) {
        return _super.call(this, props) || this;
    }
    CategorySelector.prototype.render = function () {
        var _a = this.props, categories = _a.categories, onCategoryChange = _a.onCategoryChange;
        return (this.props.categories && (React.createElement(semantic_ui_react_1.Dropdown, { fluid: true, onChange: onCategoryChange, options: tslib_1.__spread(categories.map(function (cat) {
                return { key: cat, text: cat, value: cat };
            })), placeholder: 'Select Category', search: true })));
    };
    CategorySelector.defaultProps = {
        categories: [],
    };
    return CategorySelector;
}(React.Component));
exports.CategorySelector = CategorySelector;
//# sourceMappingURL=CategorySelector.js.map