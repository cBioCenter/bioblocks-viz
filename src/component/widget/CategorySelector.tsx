import * as React from 'react';
import { Dropdown } from 'semantic-ui-react';

export interface ICategorySelectorProps {
  /** Categories to select from. */
  categories: string[];
  /** Callback for when a new category is selected. */
  onCategoryChange?(event: React.SyntheticEvent<any>, data: object): void;
}

/**
 * Class to represent a dropdown.
 *
 * @extends {React.Component<ICategorySelectorProps, any>}
 */
export class CategorySelector extends React.Component<ICategorySelectorProps, any> {
  public static defaultProps = {
    categories: [],
  };

  constructor(props: ICategorySelectorProps) {
    super(props);
  }

  public render() {
    const { categories, onCategoryChange } = this.props;

    return (
      this.props.categories && (
        <Dropdown
          fluid={true}
          onChange={onCategoryChange}
          options={[
            ...categories.map(cat => {
              return { key: cat, text: cat, value: cat };
            }),
          ]}
          placeholder={'Select Category'}
          search={true}
        />
      )
    );
  }
}
