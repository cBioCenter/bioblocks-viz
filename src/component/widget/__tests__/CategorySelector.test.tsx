import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import * as React from 'react';

import { Dropdown, DropdownItem } from 'semantic-ui-react';
import CategorySelector from '../CategorySelector';

describe('CategorySelector', () => {
  const categories = [
    'Costello Music',
    'Here We Stand',
    'We Need Medicine',
    'Eyes Wide, Tongue Tied',
    'In Your Own Sweet Time',
  ];

  it('Should match existing snapshot when given simple data.', () => {
    const selectorWrapper = shallow(<CategorySelector categories={categories} />);
    expect(toJson(selectorWrapper)).toMatchSnapshot();
  });

  it('Should invoke callback if provided when category is selected', () => {
    const onCategoryChangeSpy = jest.fn();
    const selectorWrapper = mount(<CategorySelector categories={categories} onCategoryChange={onCategoryChangeSpy} />);

    const selectionIndex = categories.indexOf('Here We Stand');
    selectorWrapper.find(Dropdown).simulate('click');
    selectorWrapper
      .find(DropdownItem)
      .at(selectionIndex)
      .simulate('click');
    expect(onCategoryChangeSpy).toHaveBeenCalled();
  });
});
