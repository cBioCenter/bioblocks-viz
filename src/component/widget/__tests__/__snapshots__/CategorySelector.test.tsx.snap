// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorySelector Should match existing snapshot when given simple data. 1`] = `
<Dropdown
  additionLabel="Add "
  additionPosition="top"
  closeOnBlur={true}
  deburr={false}
  fluid={true}
  icon="dropdown"
  minCharacters={1}
  noResultsMessage="No results found."
  openOnFocus={true}
  options={
    Array [
      Object {
        "key": "Costello Music",
        "text": "Costello Music",
        "value": "Costello Music",
      },
      Object {
        "key": "Here We Stand",
        "text": "Here We Stand",
        "value": "Here We Stand",
      },
      Object {
        "key": "We Need Medicine",
        "text": "We Need Medicine",
        "value": "We Need Medicine",
      },
      Object {
        "key": "Eyes Wide, Tongue Tied",
        "text": "Eyes Wide, Tongue Tied",
        "value": "Eyes Wide, Tongue Tied",
      },
      Object {
        "key": "In Your Own Sweet Time",
        "text": "In Your Own Sweet Time",
        "value": "In Your Own Sweet Time",
      },
    ]
  }
  placeholder="Select Category"
  renderLabel={[Function]}
  search={true}
  searchInput="text"
  selectOnBlur={true}
  selectOnNavigation={true}
  wrapSelection={true}
/>
`;
